docs/DeveloperGuide.md

```diff
@@ -943,6 +982,62 @@ The above sequence diagram omits details such as internal method calls to `GetBu
 The only difference between the negative and non-negative messages is that the negative message places the `-` symbol before the `$` so `-$xx.xx` whereas the non-negative message does not contain the `-` symbol
 </div>
 
+#### Summary Statistics
+
+**Overview**
+
+The `summary` command creates a pop-up summary window that contains summary plots to provide some basic analytics about
+the user's expense data. 
+
+The activity diagram of the `summary` command is given below:
+
+<img src="images/unicash/SummaryActivityDiagram.png" width="700" />
+
+Let's first go over the sequence of events that occur within the Ui component first to understand the code that performs
+the activities shown in the activity diagram above. The sequence diagram below documents this:
+
+<img src="images/unicash/SummaryUiSequenceDiagram.png" width="700" />
+
+**Details**
+
+1. The user runs the "summary" command
+2. The command is parsed by the `Logic` component, which is explained in another sequence diagram below, and returns a
+   CommandResult object.
+3. `MainWindow` updates the `ResultDisplay` object with the `CommandResult#getFeedbackToUser` method.
+4. `MainWindow` retrieves the updated expense summary per category and per year-month from the `Logic` component. It
+   then calls `setPieChart` and `setLineChart` on the result to update the pie chart and line chart data (stored within
+   `SummaryWindow`).
+5. If the `commandResult` object's `showSummary` attribute is `true`, then it indicates that the summary window should
+   open. Subsequently, `MainWindow` performs a self-invocation by calling its own `handleSummary` method. Otherwise,
+   the execution ends.
+6. Under the `handleSummary` method:
+   1. `MainWindow` checks to see if `SummaryWindow` is already opened by calling `SummaryWindow#isShowing`. If this is
+      false, then `SummaryWindow#focus` is called.
+   2. Otherwise, `Logic#getExpenseSummaryPerCategory` is called to retrieve the expense summary per category formatted
+      in a `HashMap<String, Double>`. The result is passed as an argument when calling the `SummaryWindow#show` method.
+   3. Within the `SummaryWindow#show` method, a check is first performed to determine if the expense summary is empty.
+      If this is true, then the summary window does not open. Otherwise, the summary window is opened.
+
+ℹ️ **Note:** Steps 1 to 5 will be called every time a new command is entered. This allows the pie chart and  line chart
+to be updated automatically every time the user makes commands that may modify the list of transactions.
+
+Next, we delve deeper into the `Logic` component, specifically when `MainWindow` executed the
+`LogicManager.execute("summary")` line of code. The sequence diagram for this portion is given below. 
+
+<img src="images/unicash/SummaryExecuteSequenceDiagram.png" width="700" />
```

Comment:
Might need to increase the width, it's hard to read the text now

Reply:
I increased it to 900. Do let me know if this helps!